org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/server/wizard/ServerConnectionModel.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2017 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.ui.internal.server.wizard;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.eclipse.equinox.security.storage.StorageException;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Display;
+import org.sonarlint.eclipse.core.SonarLintLogger;
+import org.sonarlint.eclipse.core.internal.server.IServer;
+import org.sonarlint.eclipse.core.internal.server.Server;
+import org.sonarlint.eclipse.core.internal.server.ServersManager;
+import org.sonarlint.eclipse.core.internal.utils.StringUtils;
+import org.sonarsource.sonarlint.core.client.api.connected.RemoteOrganization;
+import org.sonarsource.sonarlint.core.client.api.util.TextSearchIndex;
+
+public class ServerConnectionModel extends ModelObject {
+
+  private static final String ERROR_READING_SECURE_STORAGE = "Error reading secure storage";
+
+  public static final String PROPERTY_CONNECTION_TYPE = "connectionType";
+  public static final String PROPERTY_SERVER_URL = "serverUrl";
+  public static final String PROPERTY_AUTH_METHOD = "authMethod";
+  public static final String PROPERTY_USERNAME = "username";
+  public static final String PROPERTY_PASSWORD = "password";
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this hard-coded password. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2068)